dor/providers/process_basic_image.py

```diff
@@ -398,6 +402,147 @@ def run(self) -> None:
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.service, UseFormat.text_coordinates],
+            mimetype=Mimetype.XML.value,
+        )
+
+        alto_xml = ImageTextExtractor(image_path=service_result_file.file_path, language=self.language).alto
+
+        alto_file_path = self.accumulator.file_set_directory / file_info.path
+        alto_file_path.write_text(alto_xml)
+
+        try:
+            tech_metadata = TechnicalMetadata.create(alto_file_path)
+        except JHOVEDocError:
+            return None
+
+        event = create_preservation_event(
+            "extract text coordinates with OCR", self.accumulator.collection_manager_email
+        )
+
+        self.accumulator.add_file(
+            ResultFile(
+                file_path=alto_file_path,
+                tech_metadata=tech_metadata,
+                file_info=file_info,
+                source_file_result=service_result_file,
+                event=event,
+            )
+        )
+
+
+@dataclass
+class ExtractImageText(Operation):
+    language: str = "eng"
+
+    def run(self) -> None:
+        file_info = FileInfo(
+            identifier=self.accumulator.file_set_identifier.identifier,
+            basename=self.accumulator.file_set_identifier.basename,
+            uses=[UseFunction.service, UseFormat.text_plain],
+            mimetype=Mimetype.TXT_UTF8.value,  # Some weirdness here
+        )
+
+        service_result_file = self.accumulator.get_file(
+            function=[UseFunction.service], format=UseFormat.image
+        )
+
+        try:
+            text_coordinates_result_file = self.accumulator.get_file(
+                function=[UseFunction.service], format=UseFormat.text_coordinates
+            )
+        except AccumulatorError:
+            text_coordinates_result_file = None
+
+        if text_coordinates_result_file:
+            alto_xml = text_coordinates_result_file.file_path.read_text()
+            alto_doc = AltoDoc.create(alto_xml)
+            plain_text = alto_doc.plain_text
+        else:
+            plain_text = ImageTextExtractor(image_path=service_result_file.file_path, language=self.language).text
+
+        text_file_path = self.accumulator.file_set_directory / file_info.path
+        text_file_path.write_text(plain_text)
+
+        try:
+            tech_metadata = TechnicalMetadata.create(text_file_path)
+        except JHOVEDocError:
+            return None
+
+        event = create_preservation_event(
+            "extract plain text with OCR", self.accumulator.collection_manager_email
+        )
+
+        self.accumulator.add_file(
+            ResultFile(
+                file_path=text_file_path,
+                tech_metadata=tech_metadata,
+                file_info=file_info,
+                source_file_result=service_result_file,
+                event=event,
+            )
+        )
+
+
+
+@dataclass
+class CreateTextAnnotations(Operation):
```

Comment:
Probably all the operations should be moved to a separate file or directory? Choosing not to worry about that in this PR, though I could if people feel it's time.